pkg/experiment/query_backend/backend.go

```diff
@@ -138,9 +138,9 @@ func (q *QueryBackend) read(
 }
 
 func (q *QueryBackend) withThrottling(fn func() (*queryv1.InvokeResponse, error)) (*queryv1.InvokeResponse, error) {
+	defer q.running.Dec()
```

Comment:
this is unrelated, but a previous query-backend based implementation hit an issue where the number of running queries kept going up once the limit was hit